avm/ptn/lz/sub-vending/main.bicep

```diff
@@ -0,0 +1,750 @@
+  example: '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/xxxxxxxxxx/providers/Microsoft.Network/virtualHubs/xxxxxxxxx/hubRouteTables/xxxxxxxxx'
+})
+@description('''Optional. The resource ID of the virtual hub route table to associate to the virtual hub connection (this virtual network). If left blank/empty the `defaultRouteTable` will be associated.
+
+- Type: String
+- Default value: `''` *(empty string)* = Which means if the parameter `virtualNetworkPeeringEnabled` is `true` and also the parameter `hubNetworkResourceId` is not empty then the `defaultRouteTable` will be associated of the provided Virtual Hub in the parameter `hubNetworkResourceId`.
+    - e.g. `/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/xxxxxxxxxx/providers/Microsoft.Network/virtualHubs/xxxxxxxxx/hubRouteTables/defaultRouteTable`.
+''')
+param virtualNetworkVwanAssociatedRouteTableResourceId string = ''
+
+@metadata({
+  example: [
+    {
+      id: '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/xxxxxxxxxx/providers/Microsoft.Network/virtualHubs/xxxxxxxxx/hubRouteTables/defaultRouteTable'
+    }
+    {
+      id: '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/xxxxxxxxxx/providers/Microsoft.Network/virtualHubs/xxxxxxxxx/hubRouteTables/xxxxxxxxx'
+    }
+  ]
+})
+@description('''Optional. An array of of objects of virtual hub route table resource IDs to propagate routes to. If left blank/empty the `defaultRouteTable` will be propagated to only.
+
+Each object must contain the following `key`:
+- `id` = The Resource ID of the Virtual WAN Virtual Hub Route Table IDs you wish to propagate too
+
+> See below [example in parameter file](#parameter-file)
+
+> **IMPORTANT:** If you provide any Route Tables in this array of objects you must ensure you include also the `defaultRouteTable` Resource ID as an object in the array as it is not added by default when a value is provided for this parameter.
+
+- Type: `[]` Array
+- Default value: `[]` *(empty array)*.
+''')
+param virtualNetworkVwanPropagatedRouteTablesResourceIds array = []
+
+@metadata({
+  example: [
+    'default'
+    'anotherLabel'
+  ]
+})
+@description('''Optional. An array of virtual hub route table labels to propagate routes to. If left blank/empty the default label will be propagated to only.
+
+- Type: `[]` Array
+- Default value: `[]` *(empty array)*.
+''')
+param virtualNetworkVwanPropagatedLabels array = []
+
+@metadata({
+  example: false
+})
+@description('''Optional. Indicates whether routing intent is enabled on the Virtual Hub within the Virtual WAN.
+
+- Type: Boolean.
+''')
+param vHubRoutingIntentEnabled bool = false
+
+@metadata({
+  example: true
+})
+@description('''Optional. Whether to create role assignments or not. If true, supply the array of role assignment objects in the parameter called `roleAssignments`.
+
+- Type: Boolean.
+''')
+param roleAssignmentEnabled bool = false
+
+@metadata({
+  example: [
```

Comment:
Please note that examples will be pull from the tests into the readme automatically so that consumers can read up on them. Perssonally I'd suggest not to add extra  manual 'examples' as metadata as those tend to run stale.